crates/starknet-server/src/api/http/endpoints/time.rs

```diff
@@ -1,13 +1,43 @@
-use axum::Json;
+use axum::{Extension, Json};
 
 use crate::api::http::error::HttpApiError;
-use crate::api::http::models::Time;
-use crate::api::http::HttpApiResult;
+use crate::api::http::models::{IncreaseTime, SetTime, Time};
+use crate::api::http::{HttpApiHandler, HttpApiResult};
 
-pub(crate) async fn set_time(Json(_data): Json<Time>) -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+pub(crate) async fn set_time(
+    Json(data): Json<Time>,
+    Extension(state): Extension<HttpApiHandler>,
+) -> HttpApiResult<Json<SetTime>> {
+    let mut starknet = state.api.starknet.write().await;
+    starknet.set_pending_block_timestamp(data.time);
+    let _ = starknet.create_block().map_err(|_| HttpApiError::CreateEmptyBlockError);
+
+    let last_block = starknet.get_latest_block();
+    match last_block {
+        Ok(block) => Ok(Json(SetTime {
+            block_timestamp: block.timestamp().0,
+            block_hash: block.block_hash(),
+        })),
+        Err(_err) => Err(HttpApiError::SetTimeError),
+    }
 }
 
-pub(crate) async fn increase_time(Json(_data): Json<Time>) -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+pub(crate) async fn increase_time(
+    Json(data): Json<Time>,
+    Extension(state): Extension<HttpApiHandler>,
+) -> HttpApiResult<Json<IncreaseTime>> {
+    let mut starknet = state.api.starknet.write().await;
+    starknet.increase_time(data.time);
+    starknet.update_pending_block_timestamp();
+    let _ = starknet.create_block().map_err(|_| HttpApiError::CreateEmptyBlockError);
```

Comment:
https://github.com/0xSpaceShard/starknet-devnet-rs/pull/217#discussion_r1372876739

Reply:
Fixed in d18ddf7bcf3740b4842af8b178833ffee7453caf